pkg/operation/botanist/kubeapiserver.go

```diff
@@ -288,6 +296,38 @@ func (b *Botanist) DeployKubeAPIServer(ctx context.Context) error {
 	return nil
 }
 
+func (b *Botanist) computeKubeAPIServerSAConfig(externalHostname string) (kubeapiserver.ServiceAccountConfig, error) {
+	var saConfig *gardencorev1beta1.ServiceAccountConfig
+	if b.Shoot.GetInfo().Spec.Kubernetes.KubeAPIServer != nil && b.Shoot.GetInfo().Spec.Kubernetes.KubeAPIServer.ServiceAccountConfig != nil {
+		saConfig = b.Shoot.GetInfo().Spec.Kubernetes.KubeAPIServer.ServiceAccountConfig.DeepCopy()
+	}
+
+	shouldManageIssuer := v1beta1helper.HasManagedIssuer(b.Shoot.GetInfo())
+	canManageIssuer := b.Garden != nil && b.Garden.ShootServiceAccountIssuerHostname != nil && len(strings.TrimSpace(*b.Garden.ShootServiceAccountIssuerHostname)) != 0
```

Comment:
Again, you could remove the `TrimSpace` and validate it when setting the field/throw an error there/early instead of "silently" doing nothing here.